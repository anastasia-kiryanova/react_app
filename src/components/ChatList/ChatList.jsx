import React from 'react'
import { nanoid } from 'nanoid'
import { List, ListItem, ListItemButton, ListItemText } from '@mui/material'

const chats = [
  {
    id: nanoid(),
    name: 'Чат 1',
  },
  {
    id: nanoid(),
    name: 'Чат 2',
  },
  {
    id: nanoid(),
    name: 'Чат 3',
  },
  {
    id: nanoid(),
    name: 'Чат 4',
  },
]

export const ChatList = () => {
  return (
    <List sx={{
          width: '250px',
          display: 'flex',
          justifyContent: 'space-between',
          flexDirection: 'column',
          bgcolor: 'primary.main',
          mt: '30px',
          padding: '12px',
          color: '#ffffff',
        }} data-testid='list'>
      {chats.map((item) => (
        <ListItem key={item.id}>
          <ListItemButton>
            <ListItemText primary={item.name} />
          </ListItemButton>
        </ListItem>
      ))}
    </List>
  )
}