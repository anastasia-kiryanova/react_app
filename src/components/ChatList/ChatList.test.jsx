import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { ChatList } from './ChatList';

describe ('ChatList', () => {
  it ('is a Component', () => {
    expect(ChatList).toBeInstanceOf(Function);
  }); 

  it ('test snapshot', () => {
    const { asFragment } = render(<ChatList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it ('renders ChatList component', () => {
    render(<ChatList />);
    expect(screen.getByTestId('list')).toBeTruthy();
  }); 
});