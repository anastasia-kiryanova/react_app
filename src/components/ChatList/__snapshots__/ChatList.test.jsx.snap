// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList test snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-76yw16-MuiList-root"
    data-testid="list"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Чат 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Чат 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Чат 3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Чат 4
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
